public/js/currentAndNext.js

```diff
@@ -0,0 +1,121 @@
+/* global MyRadioAPIKey */
+$(document).ready(function() {
+
+  function scheduleUpdate() {
+    // Call the function 10 seconds past every half hour
+    let nextCall = new Date();
+
+    if (nextCall.getMinutes() >= 30) {
+      nextCall.setHours(nextCall.getHours() + 1);
+      nextCall.setMinutes(0);
+    } else {
+      nextCall.setMinutes(30);
+    }
+    nextCall.setSeconds(10);
+
+    let difference = nextCall - new Date();
+    setTimeout(updateShow, difference);
+  }
+
+  // Used for autoupdating the now and next.
+  function updateShow() {
+    var data;
+    $.ajax({
+      dataType: "json",
+      url: "//ury.org.uk/api/v2/timeslot/currentandnext/?api_key=" + MyRadioAPIKey,
+      data,
+      success(data) {
+        var calcTime = function(timestamp) {
+          var date = new Date(timestamp * 1000);
+          // Hours part from the timestamp
+          var hours = "0" + date.getHours();
+          // Minutes part from the timestamp
+          var minutes = "0" + date.getMinutes();
+          // Use substr to remove the extra 0 if 2 digit hour/min
+          return hours.substr(-2) + ":" + minutes.substr(-2);
+        };
+
+        var makeContent = function(show) {
+          if (typeof show.title !== "undefined") {
+            // If we're off air
+            if (show.end_time === "The End of Time") {
+              return "<h5 class=\"ellipsis\">We're off air right now</h5>" +
+                "<h6>Check back next term</h6>";
+            }
+            // Use "Now" as start time if it's missing
+            let startTimeString = calcTime(show.start_time);
+            if (typeof show.start_time === "undefined") {
+              startTimeString = "Now";
+            }
+            // Default case (regular show)
+            return "<h5 class=\"ellipsis\">" +
+              show.title +
+              "</h5>" +
```

Comment:
All of the headings in this will now need updating to match the new headings

Reply:
👍